backend/src/main/java/com/redhat/cloud/notifications/db/BehaviorGroupResources.java

```diff
@@ -55,30 +70,83 @@
+        return this.update(accountId, behaviorGroup, false);
+    }
+
+    public Uni<Boolean> updateDefault(BehaviorGroup behaviorGroup) {
+        return this.update(null, behaviorGroup, true);
+    }
+
+    private Uni<Boolean> update(String accountId, BehaviorGroup behaviorGroup, boolean isDefaultBehavior) {
         String query = "UPDATE BehaviorGroup SET displayName = :displayName WHERE accountId = :accountId AND id = :id";
         return sessionFactory.withSession(session -> {
-            return session.createQuery(query)
-                    .setParameter("displayName", behaviorGroup.getDisplayName())
-                    .setParameter("accountId", accountId)
-                    .setParameter("id", behaviorGroup.getId())
-                    .executeUpdate()
-                    .call(session::flush)
-                    .onItem().transform(rowCount -> rowCount > 0);
+            return this.getBehaviorGroup(session, behaviorGroup.getId(), isDefaultBehavior)
+                    .replaceWith(
+                            session.createQuery(query)
+                            .setParameter("displayName", behaviorGroup.getDisplayName())
+                            .setParameter("accountId", accountId)
+                            .setParameter("id", behaviorGroup.getId())
+                            .executeUpdate()
+                            .call(session::flush)
+                            .onItem().transform(rowCount -> rowCount > 0)
+                    );
         });
     }
 
-    // TODO Should this be forbidden for default behavior groups?
     public Uni<Boolean> delete(String accountId, UUID behaviorGroupId) {
+        return this.delete(accountId, behaviorGroupId, false);
+    }
+
+    public Uni<Boolean> deleteDefault(UUID behaviorGroupId) {
+        return this.delete(null, behaviorGroupId, true);
+    }
+
+    public Uni<Boolean> delete(String accountId, UUID behaviorGroupId, boolean isDefaultBehavior) {
         String query = "DELETE FROM BehaviorGroup WHERE accountId = :accountId AND id = :id";
         return sessionFactory.withSession(session -> {
-            return session.createQuery(query)
-                    .setParameter("accountId", accountId)
-                    .setParameter("id", behaviorGroupId)
-                    .executeUpdate()
-                    .call(session::flush)
-                    .onItem().transform(rowCount -> rowCount > 0);
+            return this.getBehaviorGroup(session, behaviorGroupId, isDefaultBehavior)
+                    .replaceWith(
+                        session.createQuery(query)
+                                .setParameter("accountId", accountId)
+                                .setParameter("id", behaviorGroupId)
+                                .executeUpdate()
+                                .call(session::flush)
+                                .onItem().transform(rowCount -> rowCount > 0)
+                    );
```

Comment:
We didn't throw an exception in case of unknown behavior group before.
```suggestion
                    )
                    .onFailure(NoResultException.class).recoverWithItem(Boolean.FALSE);
```